lib/internal/bootstrap/switches/does_own_process_state.js

```diff
@@ -107,6 +108,10 @@ function wrapPosixCredentialSetters(credentials) {
 // directory is changed by `chdir`, it'll be updated.
 let cachedCwd = '';
 
+function resetCwdCache() {
+  cachedCwd = '';
```

Comment:
Actually, I think we only need to add the reset logic to `wrappedCwd()` below in a serialize callback. That's also e.g. how we clean up stdio streams.

```JS
function wrappedCwd() {
  if (cachedCwd === '') {
    cachedCwd = rawMethods.cwd();
    if (isBuildingSnapshot()) {
      addSerializeCallback(() => { cachedCwd = ''; })
    }
  }
  return cachedCwd;
}
```